llvm/cmake/modules/HandleLLVMOptions.cmake

```diff
@@ -309,15 +309,15 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "AIX")
     message(WARNING "Build and install environment path info may be exposed; binaries will also be unrelocatable.")
   endif()
 endif()
-
+#ifndef EMSCRIPTEN
 # Pass -Wl,-z,defs. This makes sure all symbols are defined. Otherwise a DSO
 # build might work on ELF but fail on MachO/COFF.
 if(NOT (CMAKE_SYSTEM_NAME MATCHES "Darwin|FreeBSD|OpenBSD|DragonFly|AIX|OS390" OR
```

Comment:
```suggestion
if(NOT (CMAKE_SYSTEM_NAME MATCHES "Darwin|FreeBSD|OpenBSD|DragonFly|AIX|OS390|Emscripten" OR
```

Reply:
There are too many `OR`s and `NOT`s for my brain but I think that's what you want.

Reply:
Maybes sense to me, with this change made.